config_system/config_system/utils.py

```diff
@@ -47,3 +50,27 @@ def open_and_write_if_changed(fname):
             logger.debug("Updating {}".format(fname))
             with open(fname, "wt") as fp:
                 fp.write(buf.getvalue())
+
+
+def get_config_dir():
```

Comment:
I wonder if we want to have these return `__args.config` unchanged? (i.e. without running `dirname`)

This would allow more flexibility in the callers - it shouldn't be a problem for them to run `os.path.dirname` themselves, and it'll be easy to get the actual filenames then if we need them later.

Reply:
I think having functions identify what the callers actually want is useful, and separates what the plugins want to know about from how the argument(s) are passed in / structured.

If something wanted to know the config filename, I would say we add a get_config_file() wrapper.